BoneReconstructionPlanner/BoneReconstructionPlanner.py

```diff
@@ -248,14 +243,15 @@ def updateGUIFromParameterNode(self, caller=None, event=None):
 
     # Update node selectors and sliders
     self.ui.fibulaLineSelector.setCurrentNode(self._parameterNode.GetNodeReference("fibulaLine"))
+    self.ui.mandibleCurveSelector.setCurrentNode(self._parameterNode.GetNodeReference("mandibleCurve"))
+    self.ui.mandibularSegmentationSelector.setCurrentNode(self._parameterNode.GetNodeReference("mandibularSegmentation"))
+    self.ui.fibulaSegmentationSelector.setCurrentNode(self._parameterNode.GetNodeReference("fibulaSegmentation"))
     
-    # Update buttons states and tooltips
-    if self._parameterNode.GetNodeReference("fibulaLine"):
-      self.ui.createPlanesButton.toolTip = "Create fibula planes from mandibular planes"
-      #self.ui.createPlanesButton.enabled = True
-    else:
-      self.ui.createPlanesButton.toolTip = "Select fibula line and mandibular planes"
-      #self.ui.createPlanesButton.enabled = False
+    self.ui.initialLineEdit.text = self._parameterNode.GetParameter("initialSpace")
+    self.ui.betweenLineEdit.text = self._parameterNode.GetParameter("betweenSpace")
+
+    self.ui.rightFibulaCheckBox.checked = bool(self._parameterNode.GetParameter("rightFibula"))

```

Comment:
What value do you set in the parameter? I would expect `bool()` will return True for any non-empty string input, such as "False" or "0".

Reply:
By using a debugger you can get definitive answer in 10 seconds.

Reply:
(I used the debugger and I couldn't find it before I asked. And I found that the line was executing right but the tick didn't change when I press the button. So that's why I asked.) I'll try again later. If I can't find the error I'll keep working on the transforms. And maybe later I have a more clear mind and I can find it.

Reply:
You add a breakpoint to the line. When the code stops there, you can run the same command in the Python console inside the debugger. If it does not work then you modify the line. For example, just print self._parameterNode.GetParameter("rightFibula"). You will then see if the parameter value is wrong or the bool casting is wrong.

I can tell that regardless of bool() works correctly, it should not be used in a situation like this, as you don't know what value of that parameter is. It is better to do something like in the scripted module template:

    self.ui.invertOutputCheckBox.checked = (self._parameterNode.GetParameter("Invert") == "true")